dluhc-component-library/src/components/maps/MapComponent.tsx

```diff
@@ -0,0 +1,73 @@
+import MapContainer from "./mapContainer";
+import BaseMap from "./baseMap";
+import DrawingLayer from "./drawingLayer";
+import { CSSProperties, useMemo } from "preact/compat";
+
+interface MapComponentProps {
+  baseMapProps?: BaseMapProps;
+  drawingMapProps?: DrawingMapProps;
+  id?: string;
+  className?: string;
+  style?: CSSProperties;
+}
+
+interface BaseMapProps {
+  lat?: number;
+  lng?: number;
+  zoom?: number;
+  drawingMode?: boolean;
+}
+
+interface DrawingMapProps {
+  strokeColor?: string;
+  fillColor?: string;
+  strokeWidth?: number;
+  circleRadius?: number;
+  circleFillColor?: string;
+}
+
+const MapComponent = ({
+  id,
+  className,
+  style = { height: "500px", width: "500px" },
+  baseMapProps = {
+    drawingMode: true,
+    lat: 54.97,
+    lng: -1.65,
+    zoom: 10,
+  },
+  drawingMapProps = {
+    strokeColor: "#ffcc33",
+    fillColor: "rgba(255, 255, 255, 0.2)",
+    strokeWidth: 2,
+    circleRadius: 7,
+    circleFillColor: "#ffcc33",
+  },
+}: MapComponentProps) => {
+  const props = useMemo(
```

Comment:
we dont need to do this, its only 3 known props, we can hand pass them through which is less work than this.

If we think its going to be a lot of values, we can just do `...remainingProps` in the deconstructor then pass that to the MapContainer